vxfreeswitch/tests/helpers.py

```diff
@@ -0,0 +1,325 @@
+                    key, value = line.split(":", 1)
+                    command[key] = value.strip()
+            cmds.append(command)
+        self.data = data
+        return cmds
+
+
+class EslTransport(StringTransport):
+
+    def __init__(self):
+        StringTransport.__init__(self)
+        self.cmds = DeferredQueue()
+        self.esl_parser = EslParser()
+
+    def write(self, data):
+        StringTransport.write(self, data)
+        for cmd in self.esl_parser.parse(data):
+            self.cmds.put(cmd)
+
+
+class RecordingServer(Protocol):
+    def __init__(self):
+        self.command_parser = EslParser()
+
+    def connectionMade(self):
+        self.factory.clients.append(self)
+
+    def _send_event(self, content):
+        self.transport.write(
+            'Content-Length: %s\n' % len(content) +
+            'Content-Type: text/event-plain\n\n' +
+            content)
+
+    def dataReceived(self, line):
+        commands = self.command_parser.parse(line)
+        #self.factory.data.extend(commands)
+        for cmd in commands:
+            response = self.factory.get_response(cmd)
+            self.transport.write(response.to_bytes())
+            continue
+            uuid = '%s' % self.factory.uuid()
+            # Send job received correctly
+            self.transport.write(
+                'Content-Type: command/reply\n'
+                'Reply-Text: +OK\n'
+                'Job-UUID: %s\n\n' % uuid)
+            if cmd.cmd_type.startswith('bgapi originate'):
+                # Send job complete success response
+                if self.factory.fail_connect:
+                    content_body = '+ERROR %s\n' % uuid
+                else:
+                    content_body = '+OK %s\n' % uuid
+                content = (
+                    'Content-Length: %d\n'
+                    'Event-Name: BACKGROUND_JOB\n'
+                    'Job-UUID: %s\n\n'
+                    '%s' %
+                    (len(content_body), uuid, content_body))
+                self._send_event(content)
+
+    def hangup(self):
+        content = (
+            'Event-Name: CHANNEL_HANGUP\n')
+        self._send_event(content)
+
+
+class FixtureResponse(object):
+    """ A response to an ESL command. """
+
+    AUTH_REQUEST = 'auth/request'
+    API_RESPONSE = 'api/response'
+    REPLY = 'command/reply'
+    EVENT = 'text/event-plain'
+
+    def __init__(self, content_type, content=None, headers=()):
+        self.content_type = content_type
+        self.content = content
+        self.headers = headers
+
+    def to_bytes(self):
+        lines = []
+        lines.append("Content-Type: %s\n" % self.content_type)
+        if self.content is not None:
+            lines.append("Content-Length: %d\n" % len(self.content))
+        lines.extend("%s: %s\n" % (k, v) for k, v in self.headers)
+        lines.append("\n")
+        if self.content is not None:
+            lines.append(self.content)
+        return "".join(lines)
```

Comment:
Wouldn't it make more sense conceptually to have `"\n".join(lines)` and for lines not to contain any newlines? Not particularly concerned for either direction.